lib/mqtt/handles/MqttHandle.js

```diff
@@ -0,0 +1,377 @@
+const DataType = require("../homie/DataType");
+const iso8601 = require("../../utils/iso8601");
+const Logger = require("../../Logger");
+const NotImplementedError = require("../../core/NotImplementedError");
+
+/**
+ * This abstract class represents a handle to an arbitrary level of the Homie MQTT hierarchy.
+ * Children must be able to provide Homie attributes and a base topic.
+ *
+ * @abstract
+ */
+class MqttHandle {
+    /**
+     * @param {object} options
+     * @param {string} options.topicName
+     * @param {string} options.friendlyName
+     * @param {import("../MqttController")} options.controller MqttController instance
+     * @param {boolean} [options.gettable] Whether this handle's root topic can be published, Default false
+     * @param {boolean} [options.settable] Whether this handle's /set topic should be subscribed to. Default false
+     * @param {string} [options.datatype] Data type for this handle, if gettable/settable
+     * @param {string} [options.format] Restrictions or options based on the given datatype
+     * @param {boolean} [options.retained] Whether to retain this property, default true
+     * @param {MqttHandle} [options.parent]
+     */
+    constructor(options) {
+        this.parent = options.parent === undefined ? null : options.parent;
+        this.controller = options.controller;
+        this.topicName = options.topicName;
+        this.friendlyName = options.friendlyName;
+        this.dataType = options.datatype;
+        this.format = options.format;
+
+        this.retained = options.retained === undefined ? true : options.retained;
```

Comment:
Since we're using node v14 and upwards, we can use the nullish coalescing operator
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

```suggestion
        this.retained = options.retained ?? true;
```

This also applies to all other instances of similar ternary usage